src/runtime_src/xocl/core/kernel.cpp

```diff
@@ -433,7 +433,12 @@ validate_cus(const device* device, unsigned long argidx, int memidx) const
 {
   XOCL_DEBUG(std::cout,"xocl::kernel::validate_cus(",argidx,",",memidx,")\n");
   xclbin::memidx_bitmask_type connections;
-  connections.set(memidx);
+  auto dev = xrt_core::get_userpf_device(device->get_handle());
+  int l_idx = dev->get_groupRange(memidx).first;
+  int h_idx = dev->get_groupRange(memidx).second;
+  for (int i = l_idx; i<=h_idx; i++)
```

Comment:
if l_idx and h_idx is -1, then you may need to insert memidx

Reply:
get_groupRange function never returns -1. Its either return the range or return the same memidx for DDR bank case.